app/merchant/views.py

```diff
@@ -247,8 +278,47 @@ def view_orders():
     else:
         status_filter = None
 
+    ratings = Ratings.query.filter_by(merchant_id=current_user.id).all()
+    rating_dict = {rating.vendor_id: rating for rating in ratings}
+
     return render_template(
         'merchant/orders.html',
         orders=orders.all(),
-        status_filter=status_filter
+        status_filter=status_filter,
+        ratings=rating_dict
     )
+
+
+@merchant.route('/orders/<int:order_id>', methods=['POST'])
+@login_required
+@merchant_required
+def review_orders(order_id):
+    order = Order.query.get(order_id)
+    star_rating = request.json['rating']
+    comment = request.json['review']
+
+    # TODO: flash on orders page
```

Comment:
Yes. This is one possible way of doing it. The main point is that you cannot have flash with an AJAX call. Some other ideas: 

- Similar to how success messages pop up on the bottom right hand corner on success of an AJAX call you could create an error message. This i think would be a simpler way to fix the issue.
- You could do simple frontend validation client side, this is the simplest way to fix it. 
```
<script>
/// consider the fact that there can be multiple submit buttons on one page. the following example is just for 1 form.
$('#submit-button-id').on('click', function() {
    if($('#input-box-id').val() === '') {
      alert('You didn't fill in anything');
    }
}
```